dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CachedFileTree.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CachedFileTree implements FileTree {
```

Comment:
Would appreciate documentation, e.g. I think this is "cached" in the sense that it already has the file content in maps and presumably we could have another implementation that does not.

Reply:
Or reading further, maybe not. Would "CachingFileTree" be more descriptive?